File: WCF/Windows SDK WCF Samples/TechnologySamples Beta 1/Basic/Service/ServiceInstanceContext/VB/service/service.vb

'  Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.IO
Imports System.ServiceModel
Imports System.Runtime.Serialization

Namespace Microsoft.ServiceModel.Samples

    ' Define a service contract.
    <ServiceContract(Namespace:="http://Microsoft.ServiceModel.Samples")> _
    Public Interface IInstanceContextCalculator

        <OperationContract()> _
        Function Add(ByVal n1 As Integer, ByVal n2 As Integer) As Integer
        <OperationContract()> _
        Function Subtract(ByVal n1 As Integer, ByVal n2 As Integer) As Integer
        <OperationContract()> _
        Function Multiply(ByVal n1 As Integer, ByVal n2 As Integer) As Integer
        <OperationContract()> _
        Function Divide(ByVal n1 As Integer, ByVal n2 As Integer) As Integer
        <OperationContract()> _
        Function GetInstanceContextInfo() As String
    End Interface

        ' Service class which implements the service contract.
    Public Class CalculatorService
        Implements IInstanceContextCalculator

        Public Function Add(ByVal n1 As Integer, ByVal n2 As Integer) As Integer _
Implements IInstanceContextCalculator.Add
            Return n1 + n2
        End Function

        Public Function Subtract(ByVal n1 As Integer, ByVal n2 As Integer) As Integer _
Implements IInstanceContextCalculator.Subtract
            Return n1 - n2
        End Function

        Public Function Multiply(ByVal n1 As Integer, ByVal n2 As Integer) As Integer _
Implements IInstanceContextCalculator.Multiply
            Return n1 * n2
        End Function

        Public Function Divide(ByVal n1 As Integer, ByVal n2 As Integer) As Integer _
        Implements IInstanceContextCalculator.Divide

            Return n1 / n2
        End Function

        ' Obtain information from the InstanceContext and return it as a multi-line string.

        Public Function GetInstanceContextInfo() As String _
         Implements IInstanceContextCalculator.GetInstanceContextInfo
            Dim info As String = ""
            Dim operContext As OperationContext = OperationContext.Current
            Dim instanceContext As InstanceContext = operContext.InstanceContext

            info += "    " + "State: " + instanceContext.State.ToString() + Environment.NewLine
            info += "    " + "ManualFlowControlLimit: " + instanceContext.ManualFlowControlLimit.ToString() + Environment.NewLine
            info += "    " + "HashCode: " + instanceContext.GetHashCode().ToString() + Environment.NewLine


            Return info
        End Function
    End Class

End Namespace

File: WCF/Windows SDK WCF Samples/TechnologySamples Beta 1/Basic/Service/ServiceInstanceContext/CS/service/service.cs

//  Copyright (c) Microsoft Corporation.  All Rights Reserved.

using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using System.ServiceModel;
using System.Runtime.Serialization;

namespace Microsoft.ServiceModel.Samples
{
    // Define a service contract.
    [ServiceContract(Namespace="http://Microsoft.ServiceModel.Samples")]
    public interface IInstanceContextCalculator
    {
        [OperationContract]
        int Add(int n1, int n2);
        [OperationContract]
        int Subtract(int n1, int n2);
        [OperationContract]
        int Multiply(int n1, int n2);
        [OperationContract]
        int Divide(int n1, int n2);
        [OperationContract]
        string GetInstanceContextInfo();
    }

    // Service class which implements the service contract.
    public class CalculatorService : IInstanceContextCalculator
    {
        public int Add(int n1, int n2)
        {
            return n1 + n2;
        }

        public int Subtract(int n1, int n2)
        {
            return n1 - n2;
        }

        public int Multiply(int n1, int n2)
        {
            return n1 * n2;
        }

        public int Divide(int n1, int n2)
        {
            return n1 / n2;
        }

        // Obtain information from the InstanceContext and return it as a multi-line string.

        public string GetInstanceContextInfo()
        {
            string info = "";

            OperationContext operationContext = OperationContext.Current;
            InstanceContext instanceContext = operationContext.InstanceContext;

            info += "    " + "State: " + instanceContext.State.ToString() + "\n";
            info += "    " + "ManualFlowControlLimit: " + instanceContext.ManualFlowControlLimit.ToString() + "\n";
            info += "    " + "HashCode: " + instanceContext.GetHashCode().ToString() + "\n";
            
         
            return info;
        }
    }
 
}
